frontend/lib/parsing/chpl.ypp

```diff
@@ -2428,8 +2431,11 @@ fn_decl_stmt:
     fp.where = wl.where;
     context->clearComments();
 
+    bool hadError =
+      $1.errorExpr != nullptr ||
+      ($3 != nullptr && $3->isErroneousExpression());
```

Comment:
Can we move this little snippet into the `ParserContext` somewhere since we write it twice?